File: app/src/main/java/me/rohank05/beat/command/commands/StopCommand.java
package me.rohank05.beat.command.commands;

import me.rohank05.beat.command.ICommand;
import me.rohank05.beat.lavaplayer.GuildMusicManager;
import me.rohank05.beat.lavaplayer.PlayerManager;

import net.dv8tion.jda.api.EmbedBuilder;
import net.dv8tion.jda.api.entities.MessageEmbed;
import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent;

public class StopCommand implements ICommand {
    @Override
    public void run(SlashCommandInteractionEvent event) {
        if (!CheckMusicPermission.checkRemainPermission(event)) return;
        if (!CheckMusicPermission.checkPermission(event)) return;
        final GuildMusicManager musicManager = PlayerManager.getINSTANCE().getMusicManager(event.getGuild());
        musicManager.scheduler.player.stopTrack();
        musicManager.scheduler.queue.clear();
        MessageEmbed embed = new EmbedBuilder().setDescription("Player Stopped and Queue Cleared").setColor(16760143).build();
        event.getInteraction().getHook().sendMessageEmbeds(embed).queue();
    }

    @Override
    public String getName() {
        return "stop";
    }
}

File: src/main/kotlin/com/therohankumar/commands/StopCommand.kt
package com.therohankumar.commands

import com.therohankumar.audio.MusicManagerRegistry
import com.therohankumar.commands.util.CommandUtil
import net.dv8tion.jda.api.events.interaction.command.SlashCommandInteractionEvent
import net.dv8tion.jda.api.interactions.commands.build.Commands
import net.dv8tion.jda.api.interactions.commands.build.SlashCommandData
import org.springframework.stereotype.Component

@Component
class StopCommand(private val registry: MusicManagerRegistry) : SlashCommand {

    override val name = "stop"

    override fun build(): SlashCommandData = Commands.slash(name, "Stop playback and clear the queue")

    override fun execute(event: SlashCommandInteractionEvent) {
        val manager = CommandUtil.requireSameVoice(event, registry) ?: return
        val guild = event.guild!!
        manager.scheduler.queue.clear()
        manager.player.stopTrack()
        guild.audioManager.closeAudioConnection()
        event.replyEmbeds(CommandUtil.successEmbed("Stopped", "Queue cleared and disconnected.")).queue()
    }
}
